src/Command/CommandBase.php

```diff
@@ -1155,4 +1155,98 @@ protected function convertEnvironmentAliasToUuid(InputInterface $input): void {
+   */
+  public static function getSiteGroupFromSshUrl($cloud_environment): string {
+    $ssh_url_parts = explode('.', $cloud_environment->sshUrl);
+    $sitegroup = reset($ssh_url_parts);
+
+    return $sitegroup;
+  }
+
+  /**
+   * @param $cloud_environment
+   *
+   * @return bool
+   */
+  protected function isAcsfEnv($cloud_environment): bool {
+    if (strpos($cloud_environment->sshUrl, 'enterprise-g1') !== FALSE) {
+      return TRUE;
+    }
+    foreach ($cloud_environment->domains as $domain) {
+      if (strpos($domain, 'acsitefactory') !== FALSE) {
+        return TRUE;
+      }
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * @param \AcquiaCloudApi\Response\EnvironmentResponse $cloud_environment
+   *
+   * @return array
+   * @throws \Acquia\Cli\Exception\AcquiaCliException
+   */
+  protected function getAcsfSites($cloud_environment): array {
+    $sitegroup = self::getSiteGroupFromSshUrl($cloud_environment);
+    $command = ['cat', "/var/www/site-php/$sitegroup.{$cloud_environment->name}/multisite-config.json"];
+    $process = $this->sshHelper->executeCommand($cloud_environment, $command, FALSE);
+    if ($process->isSuccessful()) {
+      return json_decode($process->getOutput(), TRUE);
+    }
+    throw new AcquiaCliException("Could not get ACSF sites");
+  }
+
+  /**
+   * @param \AcquiaCloudApi\Response\EnvironmentResponse $cloud_environment
+   *
+   * @return array
+   * @throws \Acquia\Cli\Exception\AcquiaCliException
+   */
+  protected function getCloudSites($cloud_environment): array {
+    $sitegroup = self::getSiteGroupFromSshUrl($cloud_environment);
+    $command = ['cat', "/var/www/site-php/$sitegroup.{$cloud_environment->name}/config.json"];
+    $process = $this->sshHelper->executeCommand($cloud_environment, $command, FALSE);
+    if ($process->isSuccessful()) {
+      return json_decode($process->getOutput(), TRUE);
+    }
+    throw new AcquiaCliException("Could not get Cloud sites");
+  }
+
+  /**
+   * @param \AcquiaCloudApi\Response\EnvironmentResponse $cloud_environment
+   *
+   * @return mixed
+   * @throws \Acquia\Cli\Exception\AcquiaCliException
+   */
+  protected function promptChooseAcsfSite($cloud_environment) {
+    $choices = [];
+    $acsf_sites = $this->getAcsfSites($cloud_environment);
+    foreach ($acsf_sites['sites'] as $domain => $acsf_site) {
+      $choices[] = "{$acsf_site['name']} ($domain)";
+    }
+    $choice = $this->io->choice('Choose a site', $choices);
+    $key = array_search($choice, $choices, TRUE);
+    $sites = array_values($acsf_sites['sites']);
+    $site = $sites[$key];
+
+    return $site['name'];
+  }
+
+  /**
+   * @param \AcquiaCloudApi\Response\EnvironmentResponse $cloud_environment
+   *
+   * @return mixed
+   * @throws \Acquia\Cli\Exception\AcquiaCliException
+   */
+  protected function promptChooseCloudSite($cloud_environment) {
+    $cloud_sites = $this->getCloudSites($cloud_environment);
+    $choices = array_keys($cloud_sites['databases']);
```

Comment:
Are we assuming that a database == a site here?

Reply:
Basically yes, a database is a site. This is enforced on the Cloud backend. The only exception to this is the "default" site, it's called by the sitegroup name in Cloud but "default" in Drupal.

We can either `ls` the site directories or read them from `config.json`. I chose to use `config.json` since that aligns with how we get metadata on ACSF apps.

Reply:
Oh... I think you are onto something. I just created a new DB in the Cloud UI and no corresponding directory was created on disk, so it looks like the directory structure mirrors what's in the codebase, not what's in Cloud config.

Ideally this is captured somewhere machine readable, otherwise we can just read the directories from disk as well.

Reply:
I don't think it's captured anywhere. I'd suggest using the Symfony Finder component to list the dirs in `docroot/sites`. Something like BLT has:

```
  /**
   * Gets an array of sites for the Drupal application.
   *
   * I.e., sites under docroot/sites, not including acsf 'g' pseudo-sites.
   *
   * @return array
   *   An array of sites.
   */
  protected function getSiteDirs() {
    $sites_dir = $this->get('docroot') . '/sites';
    $sites = [];

    // If BLT's template has not yet been rsynced into the project root, it is
    // possible that docroot/sites does not exist.
    if (!file_exists($sites_dir)) {
      return $sites;
    }

    $finder = new Finder();
    $dirs = $finder
      ->in($sites_dir)
      ->directories()
      ->depth('< 1')
      ->exclude(['g', 'settings'])
      ->sortByName();
    foreach ($dirs->getIterator() as $dir) {
      $sites[] = $dir->getRelativePathname();
    }

    return $sites;
  }
```